### ISSUE TITLE
How to set reverse proxy for AdGuardHome Dashboard with nginx correctly?

### ISSUE THREAD

**Issue by SongzhouYang** [AdguardTeam/AdGuardHome#4266]
### Issue Details

* **Version of AdGuard Home server:**
  * v0.107.3
* **How did you install AdGuard Home:**
  * GitHub releases
* **How did you setup DNS configuration:**
  * System
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:**
  * AMD64
* **Operating system and version:**
  * Ubuntu 20.04
  

### Expected Behavior

Show login interface.

### Actual Behavior

Get a blank page

### Screenshots

<details><summary>Screenshot:</summary>

</details>

### Additional Information

I set adguardhome to monitor the LOOPBACK with web UI at `http://127.0.0.1:8000` after config it through `http://IP:3000` after installation.

Then I use Nginx to reverse proxy 127.0.0.1:8000 with configs like

```nginx
  location / {
    proxy_set_header Host $host;
    proxy_pass http://127.0.0.1:8000/;
  }
```

when I access the web UI through `https://mydomain`, I get redirected to `https://mydomain/login.html`, then I get a blank page.

Inspect the Network through the dev tool, there is a js file named like `https://mydomain/login.5d2bff792768b88ccf06.js` cannot be accessed.

How should I config the Nginx reverse proxy to make it work.

**Comment by baoang** [comment]
Not very sure about the reverse proxy config, but I accidentally made it partially right. Accidentally and partially, because I couldn't make it right concerning the /dns-query part even when the web ui part is correct.

https://www.reddit.com/r/Adguard/comments/so3pv5/what_is_the_purpose_of_setting_dnsquery_block_in/

Maybe this can help you a bit?

```
server {
    listen                                 443 ssl http2;
    server_name                            agh.mydomain.tld;

    access_log                             /var/log/nginx/agh.log combined;
    error_log                              /var/log/nginx/agh.error.log error;
 
    ssl_certificate                        /path/to/ssl/cert;
    ssl_certificate_key                    /path/to/ssl/key;
    ssl_protocols                          TLSv1.3;
    ssl_prefer_server_ciphers              off;
    ssl_stapling                           on;
    ssl_stapling_verify                    on;
    ssl_trusted_certificate                /path/to/ssl/trusted/cert;

    location / {
        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header                   Host $host;
        proxy_set_header                   X-Real-IP $remote_addr;
        proxy_pass                         http://127.0.0.1:8080; # to avoid port conflict, changing from 80 to 8080
        proxy_redirect                     / /;
        proxy_cookie_path                  / /;
    }
}
```

**Comment by EugeneOne1** [comment]
@SongzhouYang, see [FAQ](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#nginx).  If it helped?

**Comment by SongzhouYang** [comment]
@baoang Thanks for you help. But it doesn't work.
@EugeneOne1 I've read the FAQ, and did find solution.

**Comment by SongzhouYang** [comment]
I reinstall AdGuardHome to monitor all network interface.

Then I can access the dashboard through ip `http://10.147.17.1:8880`.

However when I use Nginx to perform reverse proxy, it fails just like `http://127.0.0.1:8880`. The js file cannot be loaded.

Here is my Nginx conf:

```nginx
server {
    server_name agh.mydomain.com;

    # Allow large attachments
    client_max_body_size 128M;

    location / {
        proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header                   Host $host;
        proxy_set_header                   X-Real-IP $remote_addr;
        proxy_pass                         http://10.147.17.1:8880;
        proxy_redirect                     / /;
        proxy_cookie_path                  / /;
    }

    location /doh {
        proxy_set_header Host            $http_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

        proxy_pass http://127.0.0.1:8443/dns-query;
    }

    listen 443 ssl http2; # managed by Certbot
    ssl_certificate /path/to/fullchain.pem; # managed by Certbot
    ssl_certificate_key /path/to/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_stapling                           on;
    ssl_stapling_verify                    on;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}

server {
    if ($host = agh.mydomain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    listen 80;
    server_name agh.mydomain.com;
    return 404; # managed by Certbot
}
```

**Comment by SongzhouYang** [comment]
Hi @EugeneOne1, can u reopen this issue?

**Comment by SongzhouYang** [comment]
Hi @Cybercave, I wonder how you config the dashboard. `https` version of dashboard doesn't work, cuz the dashboard is not encrypted.

**Comment by ainar-g** [comment]
>  Inspect the Network through the dev tool, there is a js file named like https://mydomain/login.5d2bff792768b88ccf06.js cannot be accessed.

@SongzhouYang, could you please provide the full screenshot of the error?  Thanks.

**Comment by SongzhouYang** [comment]
Hi @ainar-g, sorry for that I've removed AGH from my server. The error page is totally blank while the js file title is red in the Network tool.

**Comment by baoang** [comment]
There are a few lines at the end of the AdGuardHome.yaml file, looking like this below.

```
log_file: ""
verbose: false
os:
  group: ""
  user: ""
  rlimit_nofile: 0
schema_version: 12
```

Songzhou, perhaps next time (actually removing AGH may not delete the whole dir from the disk) you can alter the values and put in the log file for reference and investigation.  My prev config file is like this.

```
log_file: "/opt/AdGuardHome/aghlog.log"
verbose: true
os:
  group: ""
  user: ""
  rlimit_nofile: 0
schema_version: 12
```

This could help AGH team to locate the problems quicker.

**Comment by SongzhouYang** [comment]
Hi all, I retrying this after I posted my last reply.

This time I succeeded and now I can access the dashboard page and also successfully set up listening to 127.0.0.1 only, as well as setting and using DOH, DOT.

I'm not really sure what the difference is between this and previous attempts, but I think it seems to be related to the local Adguard client on my MBP, since this time I closed the local Adguard client. But I'm not quite sure, because I forget if I tried to close the Adguard client in my previous attempts. And now when I reopened the Adguard client it did not reproduce the previous problem.

Thanks for all your help. Maybe I can now even answer @baoang 's problem about not being able to enable DOH successfully, cuz I have encountered and solved many problems in my recent attempts LOL.

**Comment by ainar-g** [comment]
@SongzhouYang, so is the issue currently fixed?  If so, can we close this issue?

**Comment by SongzhouYang** [comment]
Yep, I will close this issue.

**Comment by SongzhouYang** [comment]
@t-e-s-tweb Sorry I missed your reply. Now I'm using Caddy instead of Nginx. The Caddy config looks like followings:
```Caddy
# FYI: I assigned the local_adguardhome_domain to 127.0.0.1 in /etc/hosts,
# and self-signed a cert for that domain.

(cert) {
  tls path/to/your/local_adguardhome_domain/cert path/to/your/local.adguardhome.domain/key
}

your.doh.domain {
  import cert
  reverse_proxy https://local_adguardhome_domain:port

  route /self-defined-uri* {
    rewrite * /dns-query{uri}
    reverse_proxy https://local_adguardhome_domain:port
  }
}
```
Not using DoT due to there need extra add-ons for Caddy to handle raw TLS traffic.

**Comment by MozerBYU** [comment]
This is my config and it has served me well.

```
server {
    # Server Settings
    listen 80;
    listen [::]:80;
    server_name sub.domain.tld;

    # Enforce HTTPS
    return 301 https://$server_name$request_uri;
}

server {
    # Server Settings
    listen 443      ssl http2;
    listen [::]:443 ssl http2;
    server_name sub.domain.tld;

    # Additional Server Settings
    server_tokens off;

    # SSL/TLS Settings
    ssl_certificate     /etc/letsencrypt/live/domain.tld/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/domain.tld/privkey.pem;

    location / {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Protocol $scheme;
	
	proxy_pass https://<ip_address>/;
    }

    # Logging Settings
    access_log /var/log/nginx/access.log;
    error_log /var/log/nginx/error.log;
    
}
```

**Comment by mercurykd** [comment]
hi, help with setup in docker case pls
I am trying to set up DoH. I register in Windows 11 the IP of my server and the template https://mydomain.com/dns-query. If you run only adguard with open port 443 of the container, then everything works. But if I hide adguard behind nginx, then only the web interface works, and dns does not work, DNS_PROBE_FINISHED_BAD_CONFIG. What am I missing?

docker compose
```
ad:
    image: adguard/adguardhome
    volumes:
        - ./config/adguard/work:/opt/adguardhome/work
        - ./config/adguard/conf:/opt/adguardhome/conf
    networks:
        default:
            ipv4_address: 10.10.0.5
ng:
    image: nginx:stable
    volumes:
        - ./config/.profile:/root/.bashrc:ro
        - ./config/nginx.conf:/etc/nginx/nginx.conf
        - ./certs/:/certs/
    ports:
        - 443:443
    networks:
        default:
            ipv4_address: 10.10.0.6
```

nginx
```
server {
        listen       443 ssl;
        server_name  mydomain.com;
        server_tokens off;

        ssl_certificate /certs/letsencrypt;
        ssl_certificate_key /certs/letsencrypt_key;

        location / {
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header X-Forwarded-Protocol $scheme;
            proxy_pass https://10.10.0.5/;
        }
    }
```

adguard.yaml
```
trusted_proxies:
    - 10.10.0.0/24
    - 127.0.0.0/8
    - ::1/128
 tls:
  enabled: true
  server_name: mydomain.com
  force_https: false
  port_https: 443
  port_dns_over_tls: 853
  port_dns_over_quic: 853
  port_dnscrypt: 0
  dnscrypt_config_file: ""
  allow_unencrypted_doh: true
```

**Comment by adityakun1992** [comment]
Had the same issue with adguard home listening on port 3000. @Cybercave 's proxy conf resolved it for me. Thanks!

This is still a weird behavior though. I don't understand why it has to do this.

**Comment by Bigsausage64** [comment]
Sorry to revive this thread but I can't figure out how to reverse-proxy DoT via Nginx Proxy Manager (which is installed on another machine than the one hosting AGH).
Currently in my setup everything is working fine, I have already reverse-proxified the Web Interface and DoH parts.
I also know that DoT is working (when not proxified), but obviously, I'd rather not directly open a port to my AGH.
I "just" can't reverse-proxy DoT and it's making me crazy. I've tried the "Stream Host" feature form NPM but no success.
I've also tried few things with standard "Proxy Hosts" but hosts stays offline and it doesn'(t work obviously.
Can anybody help me ? Thanks for your help. Let me know what information would be useful to troublehsoot.
